source/extensions/access_loggers/open_telemetry/access_log_impl.cc

```diff
@@ -90,8 +90,15 @@ void AccessLog::emitLog(const Formatter::HttpFormatterContext& log_context,
   }
   const auto formatted_attributes = attributes_formatter_->format(log_context, stream_info);
   *log_entry.mutable_attributes() = formatted_attributes.values();
-  *log_entry.mutable_trace_id() =
-      absl::HexStringToBytes(log_context.activeSpan().getTraceIdAsHex());
+
+  // Setting the trace id if available.
+  std::string trace_id_hex = log_context.activeSpan().getTraceIdAsHex();
+  if (trace_id_hex.size() == 32) {
+    *log_entry.mutable_trace_id() = absl::HexStringToBytes(trace_id_hex);
+  } else if (trace_id_hex.size() == 16) {
+    auto trace_id = absl::StrCat(Hex::uint64ToHex(0), trace_id_hex);
+    *log_entry.mutable_trace_id() = absl::HexStringToBytes(trace_id);
+  }
```

Comment:
Could you add some comments to explain these logic? Thanks.

/wait

Reply:
update